core/src/main/java/com/homeaway/streamplatform/streamregistry/health/CompactedEventStoreHealthCheck.java

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2018-Present Expedia Group.
+ * All rights reserved.  http://www.homeaway.com
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *      http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.homeaway.streamplatform.streamregistry.health;
+
+import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
+
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.health.HealthCheck;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.KafkaAdminClient;
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.kafka.common.config.ConfigResource.Type;
+
+import com.homeaway.streamplatform.streamregistry.configuration.StreamRegistryConfiguration;
+
+/**
+ * StreamRegistry uses Kafka as its Data-store.
+ * Kafka Topic's default retention setting is "cleanup.policy=delete(7 days)", and any message older than 7 days would be
+ * deleted. So, it is necessary to make sure we periodically verify the availability of topic config "cleanup.policy=compact",
+ * and monitor this metric.
+ *
+ * This HealthCheck returns "Unhealthy" if the configuration "cleanup.policy=compact" is not available for the Kafka Topic,
+ * and return the metric "app.is_kafka_topic_compacted=2" at http://HOST:8081/private/metrics
+ *
+ */
+@Slf4j
+public class CompactedEventStoreHealthCheck extends HealthCheck {
+    private static final String METRIC_IS_KAFKA_TOPIC_COMPACTED = "app.is_kafka_topic_compacted";
```

Comment:
Is there a reason the metric prefix is just `app` rather than `streamregistry.` or even prefixed with a package name of `com.homeaway.streamplatform.streamregistry.health.`?

Reply:
The "app." syntax is needed for our DD agents to pick up the metrics